ts/pages/mesh/liquidity_market.tsx

```diff
@@ -0,0 +1,419 @@
+        setBids([]);
+        setAsks([]);
+        setSpread(undefined);
+    };
+
+    useEffect(() => {
+        const fetchData = async () => {
+            const market = markets[selectedMarketIdx];
+            const orders = await fetchOrders(market.baseAsset, market.quoteAsset);
+
+            const _bids = parseOrders(orders.bids.records, 'bid');
+            const _asks = parseOrders(orders.asks.records, 'ask');
+
+            const midPrice = calcMidPrice(_bids, _asks);
+
+            calcSlippage(_bids, midPrice);
+            calcSlippage(_asks, midPrice);
+
+            const _spread = calcSpread(_bids, _asks);
+
+            // Reverse asks for display (price descending order)
+            _asks.reverse();
+
+            setBids(_bids);
+            setAsks(_asks);
+            setSpread(_spread);
+        };
+
+        resetData();
+
+        // tslint:disable-next-line:no-floating-promises
+        fetchData();
+    }, [selectedMarketIdx]);
+
+    const parseOrders = (records: OrderbookRecord[], op: Operation) => {
+        const { getPrice, getSize, orderSizeThreshold = 0 } = markets[selectedMarketIdx];
+
+        const mappedRecords = records.map(({ order }) => ({
+            order,
+            price: getPrice(order, op),
+            size: getSize(order, op),
+        }));
+
+        const filteredRecords = mappedRecords.filter(order => order.size.gte(orderSizeThreshold));
+
+        // if there are more filtered records than required, return them
+        if (filteredRecords.length >= ORDER_COUNT) {
+            return filteredRecords.slice(0, ORDER_COUNT);
+        }
+
+        // else fall back to all retrieved records and return first
+        return mappedRecords.slice(0, ORDER_COUNT);
+    };
+
+    const calcMidPrice = (_bids: OrderData[], _asks: OrderData[]) => {
+        const highestBidPrice = _bids[0].price;
+        const lowestAskPrice = _asks[0].price;
+
+        return highestBidPrice.plus(lowestAskPrice).dividedBy(2);
+    };
+
+    const calcSlippage = (orders: OrderData[], midPrice: BigNumber) => {
+        return orders.map(order => {
+            order.slippage = order.price.minus(midPrice).dividedBy(midPrice).multipliedBy(100);
+            return order;
+        });
+    };
+
+    const calcSpread = (_bids: OrderData[], _asks: OrderData[]) => {
+        const highestBidSlippage = _bids.reduce((max, order) => {
+            return max.lt(order.slippage) ? order.slippage : max;
+        }, _bids[0].slippage);
+
+        const lowestAskSlippage = _asks.reduce((min, order) => {
+            return min.gt(order.slippage) ? order.slippage : min;
+        }, _asks[0].slippage);
+
+        return highestBidSlippage.minus(lowestAskSlippage).abs();
+    };
+
+    const handleSelectItem = (index: number) => {
+        setSelectedMarketIdx(index);
+    };
+
+    return (
+        <>
+            <LiquidityMarketSelect items={markets} selectedItem={selectedMarketIdx} onSelectItem={handleSelectItem} />
```

Comment:
not sure whether this is the place nor what the size specs is, but I'd like to have the LiquidityMarketSelect text a bit smaller